Source/WebCore/page/scrolling/ScrollAnchoringController.cpp

```diff
@@ -50,35 +53,138 @@ ScrollOffset ScrollAnchoringController::computeOffset(RenderBox& candidate)
     return ScrollOffset(FloatPoint(candidate.absoluteBoundingBoxRect().location() - m_frameView.layoutViewportRect().location()));
 }
 
-void ScrollAnchoringController::chooseAnchorElement()
+static Element* anchorElementForPriorityCandidate(Element* element)
 {
-    // TODO: fully implement anchor node selection algorithm in spec
-    // (specifically excluded subtrees and priority nodes)
-    // https://drafts.csswg.org/css-scroll-anchoring/#anchor-node-selection
+    while (element) {
+        if (auto renderer = element->renderBox()) {
+            if (!renderer->isAnonymousBlock() && (!renderer->isInline() || renderer->isAtomicInlineLevelBox()))
+                return element;
+        }
+        element = element->parentElement();
+    }
+    return nullptr;
+}
 
-    auto* document = m_frameView.frame().document();
-    if (!document)
-        return;
+bool ScrollAnchoringController::examinePriorityCandidate(Element* element)
+{
+    if (auto* candidateElement = anchorElementForPriorityCandidate(element)) {
+        LOG_WITH_STREAM(ScrollAnchoring, stream << "ScrollAnchoringController::examinePriorityCandidate()");
+        auto* iterElement = candidateElement;
```

Comment:
Our smart pointer rules say that iterElement should be a RefPtr.

Reply:
Fixed.